### PR TITLE
docs(docs-infra): Use shiki for code highlighting

### PR DESCRIPTION
**Pull request by JeanMeche**
Fixes 

* #52654
* #56873
* #54716

Also thank you to @eneajaho to land the @let support for shiki in 
* shikijs/textmate-grammars-themes#73

### CODE REVIEW

**Comment by JeanMeche** [comment]
This is pending angular/dev-infra#2189

File: adev/src/app/features/references/cli-reference-details-page/cli-reference-details-page.component.scss [hunk 1]
```diff
@@ -108,14 +108,14 @@
     &:hover {
       color: var(--primary-contrast);
       background: var(--septenary-contrast);
-    }
+    }TabApi
```
Comment (atscott) [inline adev/src/app/features/references/cli-reference-details-page/cli-reference-details-page.component.scss:111]: Is this a mistake?

File: packages/common/src/pipes/slice_pipe.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ import {invalidPipeArgumentError} from './invalid_pipe_argument_error';
  *
  * produces the following:
  *
- * ```html
```
Comment (atscott) [inline packages/common/src/pipes/slice_pipe.ts:37]: Hmm, I'm not sure about making this change in the jsdocs. The difference makes editors (at least vscode) fail to highlight the code at all:
image
 versus 
image

It's at least a tradeoff that we'd have to consider if we can't do it as a transform when extracting the API docs.

Comment (JeanMeche) [reply packages/common/src/pipes/slice_pipe.ts:37]: Good point there is no value to use `angular-html` if there is no @-block.

**Review by devversion — APPROVED** [review]
LGTM for dev-infra

**Review by atscott — APPROVED** [review]
LGTM, now that's 🔥 Great work!

**Review by thePunderWoman — APPROVED** [review]

**Comment by atscott** [comment]
This PR was merged into the repository by commit a7b973eac57bbe4d7590dc84ca81900e97f4b704.

The changes were merged into the following branches: main, 18.1.x

**Pull request closed without merging**